File: apps/web/src/shared/constants/secret.ts
export const SECRET_SCOPE = {
  SUBSCRIBE_RESOURCES: "subscribe_resources",
  REPORT_TELEMETRY: "report_telemetry",
  SYNC_INVENTORY: "sync_inventory",
} as const

export type SecretScope = (typeof SECRET_SCOPE)[keyof typeof SECRET_SCOPE]

export const CONNECTION_SECRET_SCOPES: SecretScope[] = [
  SECRET_SCOPE.SUBSCRIBE_RESOURCES,
  SECRET_SCOPE.REPORT_TELEMETRY,
  SECRET_SCOPE.SYNC_INVENTORY,
]

export const SECRET_SCOPE_OPTIONS: Array<{
  value: SecretScope
  label: string
  description: string
}> = [
  {
    value: SECRET_SCOPE.SUBSCRIBE_RESOURCES,
    label: "Subscribe resources",
    description: "Pull shared Agents, Skills, Plugins, and workflows.",
  },
  {
    value: SECRET_SCOPE.REPORT_TELEMETRY,
    label: "Report telemetry",
    description: "Send usage and performance events to Conductor.",
  },
  {
    value: SECRET_SCOPE.SYNC_INVENTORY,
    label: "Sync inventory",
    description: "Synchronize the member's local EvoFlux inventory.",
  },
]

File: crates/conductor-domain/src/secret.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use std::fmt;
use std::str::FromStr;
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConnectionSecret {
    pub id: Uuid,
    pub name: String,
    pub prefix: String,
    pub owner_user_id: Uuid,
    pub scopes: Vec<SecretScope>,
    pub last_used_at: Option<DateTime<Utc>>,
    pub expires_at: Option<DateTime<Utc>>,
    pub revoked_at: Option<DateTime<Utc>>,
    pub created_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum SecretScope {
    /// Pull shared Agents, Skills and Plugin catalogs.
    SubscribeResources,
    /// Push local usage / performance telemetry.
    ReportTelemetry,
    /// Bidirectional sync of member workspace inventory.
    SyncInventory,
}

impl SecretScope {
    pub const ALL: [Self; 3] = [
        Self::SubscribeResources,
        Self::ReportTelemetry,
        Self::SyncInventory,
    ];

    pub fn as_str(self) -> &'static str {
        match self {
            Self::SubscribeResources => "subscribe_resources",
            Self::ReportTelemetry => "report_telemetry",
            Self::SyncInventory => "sync_inventory",
        }
    }

    pub fn parse(value: &str) -> Option<Self> {
        match value {
            "subscribe_resources" => Some(Self::SubscribeResources),
            "report_telemetry" => Some(Self::ReportTelemetry),
            "sync_inventory" => Some(Self::SyncInventory),
            _ => None,
        }
    }
}

#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub struct ParseSecretScopeError;

impl fmt::Display for ParseSecretScopeError {
    fn fmt(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
        formatter.write_str("unknown secret scope")
    }
}

impl std::error::Error for ParseSecretScopeError {}

impl FromStr for SecretScope {
    type Err = ParseSecretScopeError;

    fn from_str(value: &str) -> Result<Self, Self::Err> {
        Self::parse(value).ok_or(ParseSecretScopeError)
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateSecretRequest {
    pub name: String,
    pub scopes: Vec<SecretScope>,
    pub expires_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreatedSecret {
    pub secret: ConnectionSecret,
    /// Raw token — shown once.
    pub token: String,
}
